pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go

```diff
@@ -138,30 +171,12 @@ func (r *ReconcileWindowsMachineConfig) Reconcile(request reconcile.Request) (re
 		return reconcile.Result{}, err
 	}
 
-	if r.cloudProvider == nil {
-		// Get cloud provider specific info.
-		// TODO: This should be moved to validation section.
-		// 		Jira story: https://issues.redhat.com/browse/WINC-277
-		if instance.Spec.AWS == nil {
-			return reconcile.Result{}, errors.New("AWS Cloud provider is nil, cannot proceed further")
-		}
-		// TODO: We assume the cloud provider secret has been mounted to "/etc/cloud/credentials` and private key is
-		// 		present at "/etc/private-key.pem". We should have a validation method which checks for the existence
-		// 		of these paths.
-		//		Jira story: https://issues.redhat.com/browse/WINC-262
-		// TODO: Add validation for the fields in the WindowsMachineConfig CRD.
-		// 		Jira story: https://issues.redhat.com/browse/WINC-279
-		r.cloudProvider, err = cloudprovider.CloudProviderFactory(os.Getenv("KUBECONFIG"),
-			// We assume the credential path is `/etc/aws/credentials` mounted as a secret.
-			wkl.CloudCredentialsPath,
-			instance.Spec.AWS.CredentialAccountID,
-			"/tmp", "", instance.Spec.InstanceType,
-			instance.Spec.AWS.SSHKeyPair, wkl.PrivateKeyPath)
-
-		if err != nil {
-			return reconcile.Result{}, fmt.Errorf("error instantiating cloud provider: %v", err)
-		}
+	if err := r.getCloudProvider(instance); err != nil {
```

Comment:
We need to initialize a new cloud provider every time `instance.Spec.AWS.SSHKeyPair` or `instance.Spec.InstanceType` changes as those input are only allowed when calling `cloudprovider.CloudProviderFactory()` and not at new instance creation.

This was the most straight forward solution to allowing those fields to change.

My use of the word `instance` might have been confusing. I meant to refer to AWS instance, a better phrasing might be `Allow using multiple VM specifications`

Reply:
I am really confused now. I understand this would allow for us to change `AWS` info in the CR but beyond that I am not sure. What does `Allow using multiple VM specifications` imply? I would want only one specification.

Reply:
One example is it allows the user to change the size of instances without removing the operator or manually killing the operator pod. Say the user mistakenly enter m5.large instead of m5.xlarge. Or it turns out the workload they are trying to deploy needs more resources than possible on the instance type.